Sk reparse

shelbykauth opened the pull request:
Parse the Symphony Model using JS instead of PHP so we don't make a hit to the server.

shelbykauth committed:
added functions

ModelWriter.ucfirst() to uppercase the first character
ModelWriter.write() to parse the form data

shelbykauth committed:
Use JS Parser instead of PHP Parser

Comment out AJAX request
Insert JS ModelWriter.write() call

shelbykauth committed:
Bug Fix and Style Edit

Edited styles to put curly bracket on its own line for functions.
Changed ModelWriter.formGetter() to ModelWriter.readForm() at
declaration

mariekauth commented:
Please test your work before re-submission. The JavaScript parsing is still not correct.

https://htmlpreview.github.io/?https://github.com/shelbykauth/web-tools/blob/sk_reparse/web-tools/model-writer.htm

Also this branch will need to rebased.

shelbykauth committed:
BugFix with arg objects

Was originally written for a different returnObject from
ModelWriter.readForm().  That function changed, so this one had to as
well.
'arg' has been replaced with 'name' for the variable containing the name
of the argument.
'for (arg in' has been replaced with 'for (arg of', grabbing the object
instead of the index in the array.
Also an extra 'var' has been added 'for (var arg of args)' because it
should have been there already.

shelbykauth commented:
Rebased, and fixed the error where the name was based on the index, and thus became 0, 1, 2 instead of id, name, whatever.  I've tested it, but please review and see if there's any other changes which need to be made.

mariekauth commented on scripts/model-writer.js:
```diff
@@ -77,6 +80,86 @@ function remove_model_property(ele){
+                var name = arg.name;
+                var type = arg.type;
+                output += nl
+                        + tab + "/** @param " + type + " $" + name + "*/" + nl
```
One small update here. can you please add a space before */

                        + tab + "/** @param " + type + " $" + name + " */" + nl

mariekauth commented on scripts/model-writer.js:
```diff
@@ -77,6 +80,86 @@ function remove_model_property(ele){
+                   + " * @author Marie Kauth <marie.kauth@gmail.com>" + nl
+                   + " */" + nl
+                   + "class " + model + nl
+		           + "{" + nl
```
Remove the new line here, because you handle it in the first argument. This is adding an extra line.

		           + "{"
instead of
		           + "{" + nl

mariekauth reviewed (changes requested):
Nice job. Just two little spacing issues for the output.

mariekauth commented:
When you do your rebase could you please make it interactive.
When you commit the changes based on the code review you can mark the message as 'Code Review'

git commit -m 'Code Review'
git fetch upstream/master
git rebase -i upstream/master

Then in vi/vim put -f (for fix-up) in front of the code review changes. This way you won't have so many commits. The commits will exist, but the commit and the associated message will be combined into the commit above it.

If you have any questions please let me know.

shelbykauth commented:
Still having difficulties with changing the commit names.  The last two are supposed to both be Code Review.  All indicating spacing issues have been dealt with.  Any other reviews before pulling?
